proxyfsd/daemon.go

```diff
@@ -45,6 +46,32 @@ func Daemon(confFile string, confStrings []string, errChan chan error, wg *sync.
 		return
 	}
 
+	// Optionally launch an embedded HTTP Server for Golang runtime access;
```

Comment:
Seems to me that you wanted to be able to disable the Liveness Checker (I really don't know why), so you included a mechanism to do so.

I should think it *ATLEAST* as desirable to be able to disable this "optional" Debug Server port. Indeed, having said it was "optional", it appears it is not.

Let me propose this compromise:
1) This embedded HTTP Server be made strictly optional (e.g. make an enabled bool config var or, alternatively, if no port is specified, the behavior is to disable it), and
2) Disable this by default.

Of course the counter proposal would be to *NOT* provide this (in checked in code), and, thus, remove it from CONFIGURING.md :-).

But lets assume we want it in... then certainly we want to ensure that e.g. test setup code either explicitly or implicitly disables it.

Make sense?

Reply:
By my reading of the code, the debug HTTP server can be disabled by setting the port to 0:

        debugServerPortAsUint16, err := confMap.FetchOptionValueUint16("ProxyfsDebug", "DebugServerPort")
        if nil != err {
                debugServerPortAsUint16 = 6058 // TODO: Eventually set it to zero                                        
        }
        if uint16(0) != debugServerPortAsUint16 {
                debugServerPortAsString := fmt.Sprintf("%d", debugServerPortAsUint16)
                logger.Infof("proxyfsd.Daemon() starting debug HTTP Server on localhost:%s", debugServerPortAsString)
                go http.ListenAndServe("localhost:"+debugServerPortAsString, nil)
        }

if "ProxyfsDebug.DebugServerPort" is not specified, then  debugServerPortAsString gets set to "6058".  But if it is specified, and it's 0, then the debug server is not started -- so i would say that it's optional.

i can change it to disabled by default and have `monitord` enable it in the proxyfs config files that it generates.  but my inclination would be to explicitly enable it in the unit tests as well, since the more elaborate ones sometimes have bugs that cause them to hang.

do you feel strongly that it should be disabled in unit tests?

also, if we disable it by default then it will not be available (by default) in the SAIO environment unless i change the default config files to enable it.

i can do that.

Do you have strong feelings about disabling it for unit tests?  or for the SAIO environment?

Reply:
OK, i discovered that it is already explicitly enabled in the SAIO environment, though at a different port number (i changed the default from 6060 to 6058 because 6060 is in all the go documentation and thus more likely to be in use already).  the comment was already there.

```
# DebugServerPort will be served by an embedded HTTP Server responding to requests                                       
# as described in package net/http/pprof documentation. If not defined, the embedded                                     
# HTTP Server is not launched.                                                                                           
#                                                                                                                        
[ProxyfsDebug]
ProfileType:     None
DebugServerPort: 6058
```

in the unit tests, daemon_test.go is the only unit test that can usefully enable the go debug server since it is the only place where `Daemon()` is invoked.  I added:
```
                "ProxyfsDebug.DebugServerPort=6058",
```
to that test.